vector-im/element-android
# [Location sharing] - Fix the memory leaks (PSG-656)

## Type of change


## Content


Fixing different memory leaks found using LeakCanary. See the commits description for more info.

## Motivation and context


Closes #6674

## Screenshots / GIFs





## Tests


To check, you need too enable LeakCanary in the project. When doing tests, check there is no leaks notified by LeakCanary and related to the location share screens.

- Go to a room
- Start a live location share
- Navigate as you want in the different screen related to the live
- Stop the live
- Share a pinned location
- Share you current location

## Tested devices

- OS version(s): Android 11

## Checklist
